src/base/Ident.zig

```diff
@@ -38,6 +38,51 @@ pub fn for_text(text: []const u8) Ident {
     };
 }
 
+/// Errors that can occur when creating an identifier from text.
+pub const Error = error{
+    /// The identifier text is empty.
+    EmptyText,
+    /// The identifier text contains null bytes.
+    ContainsNullByte,
+    /// The identifier text contains control characters that could cause issues.
+    ContainsControlCharacters,
+};
+
+/// Create a new identifier from a byte slice with validation.
+/// Returns an error if the bytes are malformed or the Ident is invalid.
+pub fn from_bytes(bytes: []const u8) Error!Ident {
+    // Validate the bytes
+    if (bytes.len == 0) {
+        return Error.EmptyText;
+    }
+
+    // Check for null bytes (causes crashes in string interner)
+    if (std.mem.indexOfScalar(u8, bytes, 0) != null) {
+        return Error.ContainsNullByte;
+    }
```

Comment:
Sitting back and looking at this again, I could probably roll this check into the `for` loop... not sure if that's best.